src/BookingForm.js

```diff
@@ -0,0 +1,195 @@
+import React, { useState } from "react";
+import SearchResults from "./SearchResults.js";
+import TableRow from "./TableRow.js";
+function BookingForm({ handleNewBooking }) {
+  const [isSubmitClicked, setSubmitClicked] = useState(false);
+  const [newBookingInfo, setNewbookingInfo] = useState({
+    id: "0",
+    title: "",
+    firstName: "",
+    surname: "",
+    email: "",
+    roomId: "",
+    checkInDate: "",
+    checkOutDate: "",
+    items: []
+  });
+
+  function handleSelect(event) {
+    let newId = 5;
+    newId++;
+    const updatedCustomerInfo = {
+      id: newId,
+      title: event.target.value
+    };
+    setNewbookingInfo(updatedCustomerInfo);
+  }
+
+  function handleChange(event) {
+    const updatedCustomerInfo = {
+      ...newBookingInfo,
+      [event.target.name]: event.target.value
+    };
+    setNewbookingInfo(updatedCustomerInfo);
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    let items = [];
+    const temp = {
+      id: newBookingInfo.id,
+      title: newBookingInfo.title,
+      firstName: newBookingInfo.firstName,
+      surname: newBookingInfo.surname,
+      email: newBookingInfo.email,
+      roomId: newBookingInfo.roomId,
+      checkInDate: newBookingInfo.checkInDate,
+      checkOutDate: newBookingInfo.checkOutDate
+    };
+    items.push(temp);
+
+    setNewbookingInfo({
+      items,
+      id: "0",
+      title: "",
+      firstName: "",
+      surname: "",
+      email: "",
+      roomId: "",
+      checkInDate: "",
+      checkOutDate: ""
+    });
+    handleNewBooking(newBookingInfo.items);
```

Comment:
So when this function is invoked it only needs to be passed your new booking - handle the updating array of bookings logic in the component where the bookings state was initialised